src/site/content/en/blog/build-for-webusb/index.md

```diff
@@ -0,0 +1,1018 @@
+length is known.
+
+If the WebUSB Platform Capability descriptor has the `iLandingPage` field set to
+a non-zero value the browser then performs a WebUSB-specific `GET_URL` request
+by issuing a control transfer with the `bRequest` set to the `bVendorCode` value
+from the platform capability descriptor and `wValue` set to the `iLandingPage`
+value. The request code for `GET_URL` (`0x02`) goes in `wIndex`:
+
+<div class="w-table-wrapper">
+  <table>
+    <tr>
+    <th>bmRequestType</th>
+    <th>bRequest</th>
+    <th>wValue</th>
+    <th>wIndex</th>
+    <th>wLength</th>
+    <th>Data (response)</th>
+    </tr>
+    <tr>
+    <td><code>0b11000000</code></td>
+    <td><code>0x01</code></td>
+    <td><code>0x0001</code></td>
+    <td><code>0x0002</code></td>
+    <td>*</td>
+    <td>The URL descriptor</td>
+    </tr>
+  </table>
+</div>
+
+Again, this request may be issued twice in order to first probe for the length
+of the descriptor being read.
+
+{% Aside %}
+Support for displaying a notification when a USB device is plugged in is not
+available yet on Android.
+{% endAside %}
+
+## Platform-specific considerations
+
+While the WebUSB API attempts to provide a consistent interface for accessing
+USB devices developers should still be aware of requirements imposed on
+applications such as a web browsers in order to access devices.
+
+### macOS
+
+Nothing special is necessary for macOS. A website using WebUSB can connect to
+the device and claim interfaces any interfaces that aren't claimed by a kernel
+driver or another application.
+
+### Linux
+
+Linux is like macOS however by default most distributions do not set up user
+accounts with permission to open USB devices. A system daemon called udev is
+responsible for assigning the user and group allowed to access a device. A rule
+such as this will assign ownership of a device matching the given vendor and
+product IDs to the `plugdev` group which is a common group for users with access
+to peripherals:
+
+```vim
+SUBSYSTEM=="usb", ATTR{idVendor}=="XXXX", ATTR{idProduct}=="XXXX", GROUP="plugdev"
+```
+
+Replace `XXXX` with the hexadecimal vendor and product IDs for your device,
+e.g. `ATTR{idVendor}=="18d1", ATTR{idProduct}=="4e11"` would match a Nexus One
+phone. These must be written without the usual "0x" prefix and all lowercase
+to be recognized correctly. To find the IDs for your device run the command line
+tool `lsusb`.
+
+This rule should be placed in a file in the `/etc/udev/rules.d` directory and
+takes effect as soon as the device is plugged in. No need to restart udev.
+
+### Android
+
+The Android platform is based on Linux but does not require any modification to
+system configuration. By default any device which does not have a driver is
+accessible to the browser as long as the user acknowledges a permission prompt.
+This is an additional step that occurs when the application calls [`open()`] on
+a device and must be repeated every time the browser reloads. In addition more
```

Comment:
Is open() the only thing that repeats or is the permission prompt repeated too?

Reply:
Permission prompt is not repeated.

Reply:
I think I asked the wrong question. What is `open()` an additional step to?

Reply:
@reillyeon Can you reply to https://github.com/GoogleChrome/web.dev/pull/4240/files/450faaa540081e3e85ab8e9a4e6eb73da93e2288#r532708188?

Reply:
If I understand correctly, Reilly was talking about an operating system permission prompt, not the browser prompt that happens only on Android when `open()` is called.

Reply:
The developer does not need to take any additional steps. The difference between Android and other platforms is that the operating system will display a prompt asking the user to allow Chrome to access the device after the user dismisses the device selection prompt shown by `requestDevice()` and when `open()` is called. This prompt only appears once per device per session so if the developer calls `open()` immediately after `requestDevice()` the user will only see it once. The user will see it again when `open()` is called in a new session for devices the site has previously been granted permission to access because Android does not remember the permission decision the way Chrome does.

Note, due to changes in Android 10 and later Chrome cannot remember USB permissions correctly either.

Reply:
Can one of you please rewrite the paragraph to accurately describes what's happening?

Reply:
How about simply adding "operating system" before "permission prompt"?

Reply:
That's a good start. Also remove the bit about `open()` being an additional step. As Reilly said, it's not an additional step.

Reply:
The text doesn't say that `open()` is an additional step. It says, "This is an additional step that occurs when the application calls `open()`".

Reply:
Is this clearer,

> By default any device which doesn't have a driver built into the operating system is available to the browser. Developers should be aware however that users will encounter an additional step when connecting to the device. Once the users selects a device in response to a call to `requestDevice()` Android will display an prompt asking whether to allow Chrome to access it. This prompt also reappears if the user returns to a site which already has permission to connect to a device and the site calls `open()`.

Reply:
I've updated paragraph with Reilly's suggestion at https://github.com/GoogleChrome/web.dev/pull/4240/commits/b4505a6726c89d2f4bf4b12c8d6e8e79c298755e.